File: Back-End/app/websockets/managers/classroom_manager.py
# IMPORTS
from typing import Dict

# MODELS
from app.websockets.models.classroom import Classroom


class ClassroomManager:
    def __init__(self):
        self.existing_classes: Dict[int, Classroom] = dict()

    def get_classroom(self, classroom_id: int):
        if classroom_id in self.existing_classes:
            return self.existing_classes[classroom_id]
        else:
            return None

    def add_classroom(self, classroom_id: int):
        if classroom_id not in self.existing_classes:
            self.existing_classes[classroom_id] = Classroom(
                classroom_id=classroom_id)

    def remove_classroom(self, classroom_id: int):
        self.existing_classes.pop(classroom_id)


class_manager = ClassroomManager()

File: Back-End/rust/src/managers/classroom_manager.rs
use std::collections::HashMap;

use crate::models::classroom::Classroom;

pub struct ClassroomManager {
    existing_classes: HashMap<String, Classroom>,
}

impl ClassroomManager {
    pub fn new() -> ClassroomManager {
        ClassroomManager {
            existing_classes: HashMap::new(),
        }
    }

    pub fn get_classroom(&self, classroom_id: &str) -> Option<&Classroom> {
        self.existing_classes.get(classroom_id)
    }

    pub fn add_classroom(&mut self, classroom_id: &str) {
        if !self.existing_classes.contains_key(classroom_id) {
            self.existing_classes
                .insert(classroom_id.to_string(), Classroom::new(classroom_id));
        }
    }

    pub fn remove_classroom(&mut self, classroom_id: &str) {
        self.existing_classes.remove(classroom_id);
    }
}
